library/leakcanary-sample/src/main/java/com/example/leakcanary/ExampleApplication.java

```diff
@@ -32,5 +33,19 @@ public static RefWatcher getRefWatcher(Context context) {
   @Override public void onCreate() {
     super.onCreate();
     refWatcher = LeakCanary.install(this);
+    enabledStrictMode();
+  }
+
+  private void enabledStrictMode() {
+    StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder() //
+        .detectAll() //
+        .penaltyLog() //
+        .penaltyDeath() //
+        .build());
+    StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder() //
+        .detectAll() //
```

Comment:
One thing to note is that this `detectAll()` call enables some Activity reference counting. In my experience it gives quite a few false positives: http://stackoverflow.com/questions/21145261/strictmode-activity-instance-count-violation-2-instances-1-expected-on-rotati

Since LeakCanary monitors Activity leaks already, I've modified my StrictMode setup to manually enable everything except for the Activity counting:

```
StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder()
        .detectLeakedClosableObjects()
        .detectLeakedRegistrationObjects()
        .detectLeakedSqlLiteObjects()
        .penaltyLog()
        .penaltyDeath()
        .build());
```